samcli/local/lambdafn/runtime.py

```diff
@@ -113,7 +115,10 @@ def invoke(self, function_config, event, debug_context=None, stdout=None, stderr
                 # If we are in debugging mode, timer would not be created. So skip cleanup of the timer
                 if timer:
                     timer.cancel()
-                self._container_manager.stop(container)
+                if extensions_preview_enabled() and shutdown:
+                    self._container_manager.stop(container, do_shutdown_event=True)
+                else:
+                    self._container_manager.stop(container)
```

Comment:
Consider adding `do_shutdown_event` even in this case. I know `do_shutdown_event` has a default of `False` but might improve readability here. No strong feelings here, just noticed the different calling pattern since it is one after the other.